plugins/navigation-files-transformer/gatsby-node.js

```diff
@@ -49,11 +45,25 @@ const onCreateNode = async (api, options) => {
       children: [],
       parent: node.id,
       internal: {
+        content: "",
         contentDigest: createContentDigest(parsedContent),
-        type: "parliamentNavigation",
+        type: "ParliamentNavigation",
       },
     }
     createNode(navigationNode)
+    createParentChildLink({ parent: node, child: navigationNode })
+  }
+}
+
+exports.setFieldsOnGraphQLNodeType = args => {
+  const {
+    type: { name, nodes },
+  } = args
+  if (name === "ParliamentNavigation") {
+    return nodes
+      .map(({ id, parent, children, internal, ...rest }) => Object.keys(rest))
```

Comment:
Can we extract this logic into a function in a separate file so we can create tests against it?

Reply:
Yeah, that ugly piece of code was borrowed from: https://github.com/diegonc/gatsby-transformer-rawjson/blob/master/src/gatsby-node.js#L34

It takes the `parliamentNavigation` node and gets the keys as an array of arrays:

```json
[["section","title","order","pages","homePage"]]
```

then since `parliamentNavigation` was already an array of values it needs to reduce it down to one array:

```json
["section","title","order","pages","homePage"]
```

then convert it into an object where each key refers to a the GraphQL JSON type:

```json
{"section":"JSON","title":"JSON","order":"JSON","pages":"JSON","homePage":"JSON"}
```

which allows us to run queries like:

```javascript
export const query = graphql`
  query MarkdownTemplateQuery($id: String!) {
    file(id: { eq: $id }) {
      id
      modifiedTime(formatString: "YYYY-MM-DD")
      name
      relativePath
      childMarkdownRemark {
        htmlAst
        tableOfContents
        timeToRead
      }
    }
    parliamentNavigation {
      pages
    }
  }
```

Where `pages` contains the entire JSON structure all the way down. 

It is sufficiently close to magic but I can pull it out into it's own file if you want.

Reply:
Extracted the logic and added tests